Frameworks/QuartzCore/CALayer.mm

```diff
@@ -833,6 +831,8 @@ - (void)removeFromSuperlayer {
     [CATransaction _removeLayer:self];
 
     pSuper->priv->removeChild(self);
+
+    [pSuper setNeedsLayout];
     [self release];
```

Comment:
Looks good, can we get a XAMLCatalog sample that would demonstrate this issue 100%--maybe on a new "Bugs" section